From 57875aa14f9a867c5665a26808128577665d1769 Mon Sep 17 00:00:00 2001
From: "Gao, HuadongX" <huadongx.gao@intel.com>
Date: Mon, 13 Nov 2023 09:56:31 +0000
Subject: [PATCH] Add removeView operate if the parent has a View.

The monkey test error as below:
java.lang.IllegalStateException: The specified child already
has a parent. You must call removeView() on the child's parent
first.

Tracked-On: OAM-113171
Signed-off-by: Gao, HuadongX <huadongx.gao@intel.com>
---
 core/java/android/widget/PopupWindow.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/java/android/widget/PopupWindow.java b/core/java/android/widget/PopupWindow.java
index e7e148abb95a..c1b9674d5f82 100644
--- a/core/java/android/widget/PopupWindow.java
+++ b/core/java/android/widget/PopupWindow.java
@@ -1526,6 +1526,9 @@ public class PopupWindow {
         final PopupBackgroundView backgroundView = new PopupBackgroundView(mContext);
         final PopupBackgroundView.LayoutParams listParams = new PopupBackgroundView.LayoutParams(
                 MATCH_PARENT, height);
+        if(contentView.getParent() != null) {
+            ((ViewGroup)contentView.getParent()).removeView(contentView);
+        }
         backgroundView.addView(contentView, listParams);
 
         return backgroundView;
-- 
2.34.1

